jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -107,7 +120,11 @@ public String toString() {
     return contextAsString();
   }
 
-  public static JaegerSpanContext contextFromString(String value)
+  public static JaegerSpanContext contextFromString(String value) {
```

Comment:
who uses this?

Reply:
Well it is `public` and this isn't the legacy client, so I feel uncomfortable deleting it or adding a required argument. It is used internally to decode a span context from a `TextMap`.

Reply:
I would deprecate this method. I think the only reason it is public is because Codecs are in a different package. Since you already have `factory.createSpanContext()` method, the only thing this is doing is parsing a string representation of the span context - arguably a function that should live in the codecs themselves, since there can be different representations like the upcoming W3C Trace Context spec. So I suggest refactoring this and moving the main logic to TextMapCodec (it can be a public static method there, always taking the factory).